### DISCUSSION TITLE
Get services from DI within AddQuartz in ConfigureServices

### DISCUSSION THREAD

**Discussion by gabriele-ricci-kyklos** [quartznet/quartznet#2102] [Q&A]
I have some services configured in the DI, e.g. some singleton objects with a factory lambda:
```C#
services
    .AddSingleton
    (provider =>
    {
        ....
        return new MyClass(....);
    });
```

Now I don't find a way to retrieve the MyClass instance from the DI within AddQuartz method:
```C#
services.AddQuartz(q =>
    {
        q.SchedulerId = "Scheduler-Core";
        
        q.UseMicrosoftDependencyInjectionJobFactory();

        // q.UseMicrosoftDependencyInjectionScopedJobFactory();
        
        q.UseSimpleTypeLoader();
        q.UseInMemoryStore();
        q.UseDefaultThreadPool(tp =>
        {
            tp.MaxConcurrency = 10;
        });

       //Here MyClass is needed
```

In AddSingleton e.g. you have a lambda with IServiceProvider from which you can call GetService<T> and retrieve what you need.
How can I do this in Quartz.NET?

**Comment by lahma** [comment]
Probably would need some API overloads and experimentation. If you come up with a solution, a PR would be great!
